src/tools/fuzzing.h

```diff
@@ -2018,12 +2026,25 @@ class TranslateToFuzzReader {
         if (!wasm.functions.empty() && !oneIn(wasm.functions.size())) {
           target = pick(wasm.functions).get();
         }
-        return builder.makeRefFunc(target->name);
+        auto type = Type(HeapType(target->sig), /* nullable = */ true);
+        return builder.makeRefFunc(target->name, type);
       }
       if (type == Type::i31ref) {
         return builder.makeI31New(makeConst(Type::i32));
       }
-      return builder.makeRefNull(type);
+      if (type.isNullable()) {
+        return builder.makeRefNull(type);
+      }
```

Comment:
Why prefer to create a null over any other constant value?

Reply:
If I understand this code correctly, `ref.func` of any function with the correct signature would also be an option for nullable typed function reference types, which you take advantage of in the next loop. But if the type is nullable, that loop will never run because you always choose `ref.null` here.